src/dataset/nuscenes_dataset.py

```diff
@@ -1,19 +1,37 @@
 import nuscenes
 from nuscenes.utils import splits
 
+from dataset import generic_dataset
 
-class NuscenesDataset:
+
+class NuscenesDataset(generic_dataset.GenericDataset):
     """
     Handles loading and splitting of Nuscenes dataset
     """
     version = '1.0'
+    sensors = ['CAM_FRONT', 'CAM_FRONT_LEFT', 'CAM_FRONT_RIGHT', 'CAM_BACK', 'CAM_BACK_LEFT', 'CAM_BACK_RIGHT']
 
     def __init__(self, subset: str, dataset_root: str, mini_version: bool):
-        self.nusc = self.__get_nuscenes_instance(subset, dataset_root, mini_version)
+        super(NuscenesDataset, self).__init__()
```

Comment:
Just FYI: Python 3 does not require passing arguments to `super()` in simple inheritance cases as this one.